src/pytest_plugins/filler/filler.py

```diff
@@ -809,3 +760,44 @@ def pytest_collection_modifyitems(config: pytest.Config, items: List[pytest.Item
                     item.add_marker(mark)
         if "yul" in item.fixturenames:  # type: ignore
             item.add_marker(pytest.mark.yul_test)
+
+
+def pytest_sessionfinish(session: pytest.Session, exitstatus: int):
+    """
+    Perform session finish tasks.
+
+    - Remove any lock files that may have been created.
+    - Generate index file for all produced fixtures.
+    - Create tarball of the output directory if the output is a tarball.
+    """
+    if xdist.is_xdist_worker(session):
+        return
+
+    output: Path = session.config.getoption("output")
+    is_output_tarball = output.suffix == ".gz" and output.with_suffix("").suffix == ".tar"
+    output_dir = strip_output_tarball_suffix(output)
+
+    if is_output_stdout(output):
+        # Don't perform any further actions if the output is stdout.
+        return
```

Comment:
...especially so this "exit early" case is more prominently at the top of the function, see #1130.